src/main/java/com/algorithmtron/service/SortCharsByFrequencyService.java

```diff
@@ -0,0 +1,57 @@
+package com.algorithmtron.service;
+
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Service
+public class SortCharsByFrequencyService {
+
+    public String frequencySort(String s) {
+        Map<Character, Integer> hm = new HashMap<>();
+        StringBuilder sb = new StringBuilder();
+
+        for (char c : s.toCharArray()) {
+            if (!hm.containsKey(c)) {
+                hm.put(c, 1);
+            } else {
+                hm.put(c, hm.get(c) + 1);
+            }
+        }
+        int max = Integer.MIN_VALUE;
+
+        for (char c : hm.keySet()) {
+            if (hm.get(c) > max) {
+                max = hm.get(c);
+            }
+        }
+        Map<Integer, String> intKeyHm = new HashMap<>();
+
+        for (char c : hm.keySet()) {
+            int freq = hm.get(c);
+            String keysOfFreq = "";
+
+            if (intKeyHm.get(freq) == null) {
+                for (  int i = 0 ; i < freq ; i++) {
+                    keysOfFreq += c;
```

Comment:
Is += safe? Could we use a better version of this?

Reply:
You're right! 

Should i convert the variable keysOfFreq to a Stringbuilder instead ?

Reply:
Since String is immutable in java, when you do a +, += or concat(String), a new String is generated. The bigger the String gets the longer it takes there forth is more to copy and more garbage is produced. I really do thing that using String builder is a better approach. :)

Reply:
i totally agree :)